labs/lab-2-javascript-game/script.js

```diff
@@ -0,0 +1,123 @@
+var userPlayer;
+var grant;
+
+window.addEventListener("load", () => {
+	document.getElementById('start-button').addEventListener('click', startGame);
+	document.getElementById('attack-button').addEventListener('click', attack);
+	document.getElementById('heal-button').addEventListener('click', heal);
+	document.getElementById('quit-button').addEventListener('click', quit);
+});
+
+function startGame() {
+	const playersName = document.getElementById('players-name').value;
+	userPlayer = new Player(playersName, 40);
+	grant = new Player('Grant', 10);
+	document.getElementById('start-div').style.display = 'none';
+	document.getElementById('main-div').style.display = 'flex';
+	document.getElementById('users-name').innerText = playersName;
+	document.getElementById('grants-name').innerText = 'Grant The Almighty Chicken';
+	document.getElementById('heals-remaining').innerText = 'Heals Left: ' + userPlayer.numberOfHeals + '/2';
+	document.getElementById('win-count').innerText = 'Wins: ' + userPlayer.wins + '/5'
+}
+
+function attack() {
+	userPlayer.damageDone();
+	document.getElementById('player-text').innerText = 'Grant The Almighty Chicken attacks! ' + userPlayer.name + ' now has ' + userPlayer.health + ' health left.';
+	grant.damageDone();
+	document.getElementById('grant-text').innerText = userPlayer.name + ' attacks! Grant The Almighty Chicken now has ' + grant.health + ' health left.';
+
+	var playerHealthBar = (userPlayer.health / 40) * 100;
+	document.getElementById('players-health').style.width = playerHealthBar + '%';
+	var grantHealthBar = (grant.health / 10) * 100;
+	document.getElementById('grants-health').style.width = grantHealthBar + '%';
+
+	
+	if (playerHealthBar < 75 && playerHealthBar > 45) {
+		document.getElementById('players-health').style.backgroundColor = "yellow";
+	} else if (playerHealthBar < 45) {
+		document.getElementById('players-health').style.backgroundColor = "red";
+	}
+
+	if (grantHealthBar < 75 && grantHealthBar > 40) {
+		document.getElementById('grants-health').style.backgroundColor = "yellow";
+	} else if (grantHealthBar < 40) {
+		document.getElementById('grants-health').style.backgroundColor = "red";
+	}
+
+	if (grant.health <= 0 && userPlayer.wins <= 5) {
+		winCount();
+	}
+
+	if (userPlayer.health <= 0) {
+		grantWinner();
+	} else if (userPlayer.wins === 5 & grant.health <= 0) {
+		userWinner();
```

Comment:
This function is massive, and so is hard to debug; but I can tell you that this never gets called; even with 5 wins, I never won the game, only the round:

<img width="871" alt="screenshot 2018-02-06 09 48 55" src="https://user-images.githubusercontent.com/150973/35865986-f66c57e6-0b23-11e8-9b77-b3a72e1c8f5a.png">



(Also you can get into some weird states:
<img width="871" alt="screenshot 2018-02-06 09 47 31" src="https://user-images.githubusercontent.com/150973/35866011-03b0d594-0b24-11e8-85bf-1ab2515e1467.png">)